integration-tests/tests/src/tests/sync/flexible.ts

```diff
@@ -2075,4 +2081,400 @@ describe("Flexible sync", function () {
       });
     });
   });
+
+  describe("Progress notification", function () {
+    this.timeout(5000);
+    before(async function (this: UserContext) {
+      // TODO: Investigate this.
+      // Use a new user for this test suite as not doing so causes some session issues in other tests
+      this.user = await this.app.logIn(Credentials.anonymous(false));
+    });
+    beforeEach(async function (this: RealmContext & UserContext) {
+      // @ts-expect-error Using an internal API
+      this.realm = new Realm({
+        schema: [Person, Dog],
+        sync: {
+          flexible: true,
+          user: this.user,
+          _sessionStopPolicy: SessionStopPolicy.Immediately,
+        },
+      });
+    });
+
+    it("only estimate callback is allowed", async function (this: RealmContext) {
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars, @typescript-eslint/no-empty-function
+      const callback = (estimate: number) => {};
+      this.realm.syncSession?.addProgressNotification(
+        Realm.ProgressDirection.Download,
+        Realm.ProgressMode.ForCurrentlyOutstandingWork,
+        callback,
+      );
+    });
```

Comment:
What is this testing specifically? Passing a single argument arrow function as callback?

Reply:
Yes, this has to do with deprecated API vs non-deprecated